ts/lib/socket.ts

```diff
@@ -76,9 +78,22 @@ export class ConstellationSocket extends EventEmitter {
 
         this.once('event:hello', () => {
             this.ready = true;
+            if (this.reconnecting) {
+                this.reconnecting = false;
+                this.emit("reopen");
+            }
             this.queue.forEach(data => this.send(data));
             this.queue = [];
         });
+        this.on("close", () => {
+            if (!this.options.autoReconnect) {
+                return;
+            }
+            this.reconnecting = true;
+            setTimeout(() => {
```

Comment:
Yea, there does need to be a method for gracefully closing the socket without reconnecting (cc @UnwrittenFun ). On the Beam site that would need to be called when the user logs out, for instance.